kv/kvserver/batcheval: TestLeaseTransferWithPipelinedWrite failed

kv/kvserver/batcheval.TestLeaseTransferWithPipelinedWrite [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5971038?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5971038?buildTab=artifacts#/) on master @ [5f854ce368440e2289af5db321f7330dbef87639](https://github.com/cockroachdb/cockroach/commits/5f854ce368440e2289af5db321f7330dbef87639):


```
=== RUN   TestLeaseTransferWithPipelinedWrite
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestLeaseTransferWithPipelinedWrite3490123133
    test_log_scope.go:80: use -show-logs to present logs inline
    cmd_lease_test.go:112: worker failed: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_NEW_LEASE_PREVENTS_TXN): "sql txn" meta={id=e6c9bcc5 key=/Table/132/1/0/1/0 pri=0.06112298 epo=0 ts=1659598480.500067056,1 min=1659598480.416811047,0 seq=3} lock=true stat=PENDING rts=1659598480.500067056,1 wto=false gul=1659598480.916811047,0
    panic.go:500: -- test log scope end --
--- FAIL: TestLeaseTransferWithPipelinedWrite (56.76s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTransferWithPipelinedWrite.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18331

kv/kvserver/batcheval.TestLeaseTransferWithPipelinedWrite [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5982578?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5982578?buildTab=artifacts#/) on master @ [d687b14e513a49f94e1862c78a7685b78f4fbad9](https://github.com/cockroachdb/cockroach/commits/d687b14e513a49f94e1862c78a7685b78f4fbad9):


```
=== RUN   TestLeaseTransferWithPipelinedWrite
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestLeaseTransferWithPipelinedWrite2936323110
    test_log_scope.go:80: use -show-logs to present logs inline
    cmd_lease_test.go:112: worker failed: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_NEW_LEASE_PREVENTS_TXN): "sql txn" meta={id=1b827ad8 key=/Table/115/1/0/1/0 pri=0.02346137 epo=0 ts=1659683477.271226262,1 min=1659683477.125156301,0 seq=3} lock=true stat=PENDING rts=1659683477.271226262,1 wto=false gul=1659683477.625156301,0
    panic.go:500: -- test log scope end --
--- FAIL: TestLeaseTransferWithPipelinedWrite (39.44s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTransferWithPipelinedWrite.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestLeaseTransferWithPipelinedWrite [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5993360?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5993360?buildTab=artifacts#/) on master @ [db95c2484a9f15c2de01466af0aae19c8ab57618](https://github.com/cockroachdb/cockroach/commits/db95c2484a9f15c2de01466af0aae19c8ab57618):


```
=== RUN   TestLeaseTransferWithPipelinedWrite
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestLeaseTransferWithPipelinedWrite2679577506
    test_log_scope.go:80: use -show-logs to present logs inline
    cmd_lease_test.go:112: worker failed: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_NEW_LEASE_PREVENTS_TXN): "sql txn" meta={id=636d0007 key=/Table/130/1/0/1/0 pri=0.00852398 epo=0 ts=1659771960.580418844,0 min=1659771960.580418844,0 seq=3} lock=true stat=PENDING rts=1659771960.580418844,0 wto=false gul=1659771961.080418844,0
    panic.go:500: -- test log scope end --
--- FAIL: TestLeaseTransferWithPipelinedWrite (55.10s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTransferWithPipelinedWrite.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestLeaseTransferWithPipelinedWrite [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6036814?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6036814?buildTab=artifacts#/) on master @ [6c24f3564e213371bbe800e9646386cdefeb1546](https://github.com/cockroachdb/cockroach/commits/6c24f3564e213371bbe800e9646386cdefeb1546):


```
=== RUN   TestLeaseTransferWithPipelinedWrite
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestLeaseTransferWithPipelinedWrite3061496584
    test_log_scope.go:80: use -show-logs to present logs inline
    cmd_lease_test.go:112: worker failed: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_NEW_LEASE_PREVENTS_TXN): "sql txn" meta={id=6c1c5ff4 key=/Table/124/1/0/1/0 pri=0.05493523 epo=0 ts=1660118565.842213510,1 min=1660118565.629899239,0 seq=3} lock=true stat=PENDING rts=1660118565.842213510,1 wto=false gul=1660118566.129899239,0
    panic.go:500: -- test log scope end --
--- FAIL: TestLeaseTransferWithPipelinedWrite (45.65s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTransferWithPipelinedWrite.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestLeaseTransferWithPipelinedWrite [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6049746?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6049746?buildTab=artifacts#/) on master @ [9bd865dadb0601059ee0f8419d1ad47fb74db80a](https://github.com/cockroachdb/cockroach/commits/9bd865dadb0601059ee0f8419d1ad47fb74db80a):


```
=== RUN   TestLeaseTransferWithPipelinedWrite
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestLeaseTransferWithPipelinedWrite1002527624
    test_log_scope.go:80: use -show-logs to present logs inline
    cmd_lease_test.go:112: worker failed: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_NEW_LEASE_PREVENTS_TXN): "sql txn" meta={id=71e65089 key=/Table/131/1/0/1/0 pri=0.01318320 epo=0 ts=1660203131.418290275,1 min=1660203131.251080030,0 seq=3} lock=true stat=PENDING rts=1660203131.418290275,1 wto=false gul=1660203131.751080030,0
    panic.go:500: -- test log scope end --
--- FAIL: TestLeaseTransferWithPipelinedWrite (56.69s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTransferWithPipelinedWrite.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestLeaseTransferWithPipelinedWrite [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6096101?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6096101?buildTab=artifacts#/) on master @ [6b7f0058325de8b8b321a08137d462c418c3047f](https://github.com/cockroachdb/cockroach/commits/6b7f0058325de8b8b321a08137d462c418c3047f):


```
=== RUN   TestLeaseTransferWithPipelinedWrite
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestLeaseTransferWithPipelinedWrite2716334196
    test_log_scope.go:80: use -show-logs to present logs inline
    cmd_lease_test.go:112: worker failed: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_NEW_LEASE_PREVENTS_TXN): "sql txn" meta={id=30cd482c key=/Table/133/1/0/1/0 pri=0.01267302 epo=0 ts=1660548155.641867712,1 min=1660548155.504547373,0 seq=3} lock=true stat=PENDING rts=1660548155.641867712,1 wto=false gul=1660548156.004547373,0
    panic.go:500: -- test log scope end --
--- FAIL: TestLeaseTransferWithPipelinedWrite (67.94s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTransferWithPipelinedWrite.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestLeaseTransferWithPipelinedWrite [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6215422?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6215422?buildTab=artifacts#/) on master @ [003c0360de8b64319b5f0f127b99be91dbdca8a3](https://github.com/cockroachdb/cockroach/commits/003c0360de8b64319b5f0f127b99be91dbdca8a3):


```
=== RUN   TestLeaseTransferWithPipelinedWrite
    test_log_scope.go:162: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestLeaseTransferWithPipelinedWrite555953228
    test_log_scope.go:80: use -show-logs to present logs inline
    cmd_lease_test.go:112: worker failed: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_NEW_LEASE_PREVENTS_TXN): "sql txn" meta={id=66015018 key=/Table/111/1/0/1/0 pri=0.06724410 epo=0 ts=1661324742.116263602,0 min=1661324742.116263602,0 seq=3} lock=true stat=PENDING rts=1661324742.116263602,0 wto=false gul=1661324742.616263602,0
    panic.go:500: -- test log scope end --
--- FAIL: TestLeaseTransferWithPipelinedWrite (37.24s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseTransferWithPipelinedWrite.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/batcheval/cmd_lease_test.go
+++ b/pkg/kv/kvserver/batcheval/cmd_lease_test.go
@@ -49,8 +49,7 @@ func TestLeaseTransferWithPipelinedWrite(t *testing.T) {
 
 	db := tc.ServerConn(0)
 
-	// More than 30 iterations is flaky under stressrace on teamcity.
-	for iter := 0; iter < 30; iter++ {
+	for iter := 0; iter < 100; iter++ {
 		log.Infof(ctx, "iter %d", iter)
 		if _, err := db.ExecContext(ctx, "drop table if exists test"); err != nil {
 			t.Fatal(err)
@@ -109,7 +108,13 @@ func TestLeaseTransferWithPipelinedWrite(t *testing.T) {
 			t.Fatal("timed out")
 		case err := <-workerErrCh:
 			if err != nil {
-				t.Fatalf("worker failed: %+v", err)
+				// We allow the transaction to run into an aborted error due to a lease
+				// transfer when it attempts to create its transaction record. This it
+				// outside of the focus of this test.
+				okErr := testutils.IsError(err, roachpb.ABORT_REASON_NEW_LEASE_PREVENTS_TXN.String())
+				if !okErr {
+					t.Fatalf("worker failed: %+v", err)
+				}
 			}
 		}
 	}

